pkg/querier/dispatcher.go

```diff
@@ -305,11 +232,9 @@ func (w *queryResponseWriter) WriteError(ctx context.Context, typ mimirpb.QueryE
 }
 
 func (w *queryResponseWriter) write(ctx context.Context, resp *frontendv2pb.QueryResultStreamRequest) error {
-	if w.querierResponseMessageBytes != nil {
```

Comment:
Note to reviewers: this check is no longer required, as all requests are Protobuf requests and therefore these observers will be populated.